lib/ReactViews/ExplorerWindow/Tabs/MyDataTab/AddData.jsx

```diff
@@ -117,6 +119,22 @@ const AddData = createReactClass({
         this.state.remoteUrl,
         this.state.remoteDataType.value
       );
+    } else if (this.state.remoteDataType.value === "json") {
+      promise = loadJson(this.state.remoteUrl).then((data) => {
+        if (data.error) {
+          return Promise.reject(data.error);
+        }
+        this.props.terria.catalog.group
+          .addMembersFromJson(CommonStrata.user, data.catalog)
+          .raiseError(this.props.terria, "Failed to load catalog from file");
+
+        return Promise.resolve();
+      });
```

Comment:
It might good if the `raiseError()` can cover `loadJson()` call so that the user gets an error pop-up if the URL failed to load. One common scenario is user copies a URL that cannot be loaded due to CORS error. It will be good to show some feedback when that happens.

Reply:
That could work, but I think a simpler option might be something like:

```js
      promise = loadJson(this.state.remoteUrl)
        .then((data) => {
          if (data.error) {
            return Promise.reject(data.error);
          }
          this.props.terria.catalog.group
            .addMembersFromJson(CommonStrata.user, data.catalog)
            .raiseError(this.props.terria, "Failed to load catalog from file");
        })
        .then(() => {
          this.props.onUrlAddFinished();
        })
        .catch((error) =>
          TerriaError.from(error).raiseError(
            this.props.terria,
            `An error occurred trying to add data from URL: ${this.state.remoteUrl}`
          )
        )
        .finally(() => {
          this.setState({
            isLoading: false
          });
        });

```

which is basically `catch`ing the `loadJson()` response and raising the error to user. We will be showing two separate errors depending on whether it is a URL failure or `addMembersFromJson` failure, but that's a good thing, I think.

Also note that, I moved the `onUrlAddFinished()` to the `then()` so that the user stays in the current upload view if an error occurred. We could also maybe use `await` style to rewrite my block of code and simplify it.

Thoughts @sixlighthouses ?

Reply:
@na9da , this is a nice solution -- TerriaError was what I needed but couldn't find.  I think the .then .catch style is fine here as the rest of the file uses it and I assume it may all get a tidy up when it gets TSified